section11/numberformat_proposed.html

```diff
@@ -202,18 +225,52 @@ <h1>PartitionNumberPattern ( _numberFormat_, _x_ )</h1>
+            1. Let _measureUnit_ be _numberFormat_.[[MeasureUnit]].
+            1. Let _mu_ be an ILD string representing _measureUnit_ in long form, which may depend on _x_ in languages having different plural forms.
+            1. Append a new Record { [[Type]]: `"unit"`, [[Value]]: _mu_ } as the last element of _result_.
+          1. Else if _p_ is equal `"currencyCode"` and _numberFormat_.[[UnitType]] is `"currency"`, then
+            1. Let _currency_ be _numberFormat_.[[Currency]].
+            1. Let _cd_ be _currency_.
+            1. Append a new Record { [[Type]]: `"currency"`, [[Value]]: _cd_ } as the last element of _result_.
+          1. Else if _p_ is equal `"currencySymbol"` and _numberFormat_.[[UnitType]] is `"currency"`, then
+            1. Let _currency_ be _numberFormat_.[[Currency]].
+            1. Let _cd_ be an ILD string representing _currency_ in short form. If the implementation does not have such a representation of _currency_, use _currency_ itself.
+            1. Append a new Record { [[Type]]: `"currency"`, [[Value]]: _cd_ } as the last element of _result_.
+          1. Else if _p_ is equal `"currencyNarrowSymbol"` and _numberFormat_.[[UnitType]] is `"currency"`, then
+            1. Let _currency_ be _numberFormat_.[[Currency]].
+            1. Let _cd_ be an ILD string representing _currency_ in narrow form. If the implementation does not have such a representation of _currency_, use _currency_ itself.
+            1. Append a new Record { [[Type]]: `"currency"`, [[Value]]: _cd_ } as the last element of _result_.
+          1. Else if _p_ is equal `"currencyName"` and _numberFormat_.[[UnitType]] is `"currency"`, then
             1. Let _currency_ be _numberFormat_.[[Currency]].
-            1. Assert: _numberFormat_.[[CurrencyDisplay]] is `"code"`, `"symbol"` or `"name"`.
-            1. If _numberFormat_.[[CurrencyDisplay]] is `"code"`, then
-              1. Let _cd_ be _currency_.
-            1. Else if _numberFormat_.[[CurrencyDisplay]] is `"symbol"`, then
-              1. Let _cd_ be an ILD string representing _currency_ in short form. If the implementation does not have such a representation of _currency_, use _currency_ itself.
-            1. Else if _numberFormat_.[[CurrencyDisplay]] is `"name"`, then
-              1. Let _cd_ be an ILD string representing _currency_ in long form. If the implementation does not have such a representation of _currency_, use _currency_ itself.
+            1. Let _cd_ be an ILD string representing _currency_ in long form, which may depend on _x_ in languages having different plural forms. If the implementation does not have such a representation of _currency_, use _currency_ itself.
             1. Append a new Record { [[Type]]: `"currency"`, [[Value]]: _cd_ } as the last element of _result_.
           1. Else,
             1. Let _literal_ be the substring of _pattern_ from position _beginIndex_, inclusive, to position _endIndex_, inclusive.
```

Comment:
We should be using "unknown", not "literal", as the type here, re https://github.com/tc39/ecma402/pull/245 (though "literal" is correct for line 282).

Reply:
My snapshot of numberformat.html was probably taken before that PR.  I pulled in upstream and integrated any relevant changes into my spec draft.